vignettes/spatial_dhmv_query.Rmd

```diff
@@ -0,0 +1,397 @@
+
+In R, we can receive further information conveniently via `ows4R`. This works via the "coverage summary" &#x2026;
+
+```{r}
+feature <- "DHMVII_DTM_1m"
+dtm_smry <- caps$findCoverageSummaryById(feature, exact = TRUE)
+dtm_smry
+```
+
+&#x2026; and the coverage description.
+
+```{r}
+# Get description from the WCS client
+dtm_des <- WCS$describeCoverage(feature)
+dtm_des
+```
+
+
+Finally, it is important to know which dimensions are available: `LON/LAT`, `x/y`, or maybe `time`?
+
+```{r}
+dtm_dims <- dtm_smry$getDimensions()
+dtm_dims
+```
+
+
+So far, so good.
+
+Now, following the tutorials to get some real data:
+
+```{r}
+x_test <- 148600
+y_test <- 208900
+radius <- 1 # (+/- m)
+
+hopo_boxed <- OWSUtils$toBBOX(
+  xmin = x_test - radius,
+  xmax = x_test + radius,
+  ymin = y_test - radius,
+  ymax = y_test + radius
+  )
+```
+
+I will write the file to disk:
+
+```{r, eval=FALSE}
+tryCatch({
+  dtm_data = dtm_smry$getCoverage(
+    bbox = hopo_boxed,
+    filename = tempfile(fileext = ".tif")
+    )
+  dtm_data
+}, error = function (err) {message(err)})
+```
+
+&#x2026; or, maybe not. The function fails, somehow.
+
+*Status quo:*
+
+-   I receive a `tif` file (because *I called it* "tif"), supposedly some kind of GeoTIFF image (because I *asked* for `image/tiff`).
+-   I also receive a couple of error messages and warnings:
+    -   *"Start tag expected, '<' not found"*
+    -   *"cannot open this file as SpatRaster:&#x2026;"*
+    -   *"not recognized as being in a supported file format. (GDAL error 4)"*
+
+Maybe most useful, `ows4R` briefly displays the URL it used to attempt data query: `https://geo.api.vlaanderen.be/DHMV/wcs?service=WCS&version=2.0.1&coverageId=DHMVII_DTM_1m&subset=x(148599,148601)&subset=y(208899,208901)&format=image/tiff&request=GetCoverage`
+
+Let's break this URL down:
+
+-   the desired `service` is indeed `WCS`,
+-   `version` is `2.0.1` (I would guess)
+-   `request`, `coverageID`, and `format` seem to be as expected
+-   there are `subset` components for the respective dimensions.
+
+Feel free to try yourself to paste the URL to a browser and download the file.
+
+Still, the downloaded file is unreadable. Subtle symptom: even if one adjusts the `subset` (i.e. `bbox` argument, above), the received tif does not change in size; the download is bbox-agnostic, so to speak.
+
+Opening the file with a text editor such as [vim](https://www.vim.org), we see an XML part, some generic binary stuff, some hints that this [should be a GML](https://en.wikipedia.org/wiki/Geography_Markup_Language).
+
+There are [sample geotiffs online](https://github.com/mommermi/geotiff_sample/blob/master/sample.tif), and they look different, or maybe not. Gml is clearly not geotiff, except maybe for the binary part (which I tried to extract, to no avail).
+
+The file does not download with an extension, and it is still unclear what it actually is.
+
```

Comment:
It shouldn't. If a browser would implement auto-inference of file types upon download, this is bound to break files (only the provider knows the true file content). I would guess smart browsers just take the package  as they get it. On the other hand, neither firefox, nor chrome, and certainly not edge are guaranteed to be well-made programs.
It might also be that the web server does stuff different depending on OS, which is no good either.
I will test around a bit with my OS's and browsers.

Reply:
(But that's how life is. I have adjusted the text.)